### DISCUSSION TITLE
UDP Port Sharing between Hysteria and Other UDP protocols

### DISCUSSION THREAD

**Discussion by rwenzlofsky** [apernet/hysteria#342] [Q&A]
Hi. I am currently working remotely from Cuba for a European company and use Hysteria on my self-hosted VPN (located in Miami). First of all, congratulations, out of all protocols I tried on my VPN server this one offers me the lowest latency. It really "feels" when using the Horizon client remote access as if I would be in Europe. With TCP protocols I always had this small delay when typing on the keyboard, but with Hysteria, I don't recognize a delay at all. Another important thing is that although the Cuban INternet provider is doing internet censorship, I have no issue bypassing when running Hysteria on port 443...which leads me to my question: I run several TCP based protocols via port sharing with sslh on port TCP 443, but I would like to do the same for UDP (as I want to run Vless via QUIC as well). Do you have any ideas on how this can be achieved? Is it possible to define an SNI in the client configuration of Hysteria? This would help me already I assume as such should be able to handle UDP traffic as well.

Any help is appreciated.

**Comment by tobyxdd** [comment]
I've never tried sslh before, but looks like it does [support UDP](https://github.com/yrutschle/sslh/blob/master/doc/config.md#udp) and can recognize QUIC traffic already.

P.S. You need to turn off obfuscation for it to recognize Hysteria connections as QUIC

**Comment by tobyxdd** [comment]
> Is it possible to define an SNI in the client configuration of Hysteria

You can use `server_name` in the client config for that.
